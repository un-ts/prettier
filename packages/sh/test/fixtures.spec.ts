import fs from 'node:fs'
import path from 'node:path'

import { format } from 'prettier'

import * as sh from 'prettier-plugin-sh'

const _dirname = import.meta.dirname

describe('parser and printer', () => {
  const fixtures = path.resolve(_dirname, 'fixtures')
  for (const relativeFilepath of fs.readdirSync(fixtures)) {
    const filepath = path.resolve(fixtures, relativeFilepath)
    const input = fs.readFileSync(filepath, 'utf8')

    const filename = path.basename(filepath)

    it(`should format ${relativeFilepath} fixtures - sh`, async () => {
      try {
        const output = await format(input, {
          filepath,
          parser: 'sh',
          plugins: [sh],
        })

        expect(output).toMatchSnapshot(relativeFilepath)
      } catch (err: unknown) {
        expect((err as Error).message.split('\n').at(0)).toMatchSnapshot(
          relativeFilepath,
        )
      }
    })

    if (filename !== 'Dockerfile' && !filename.endsWith('.Dockerfile')) {
      return
    }

    it(`should format ${relativeFilepath} fixtures - dockerfile`, async () => {
      try {
        const output = await format(input, {
          filepath,
          parser: 'dockerfile',
          plugins: [sh],
        })

        expect(output).toMatchSnapshot(relativeFilepath)
      } catch (err: unknown) {
        expect((err as Error).message.split('\n').at(0)).toMatchSnapshot(
          relativeFilepath,
        )
      }
    })
  }
})
