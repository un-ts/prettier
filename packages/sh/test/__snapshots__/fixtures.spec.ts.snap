// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser and printer > should format .dockerignore fixtures - sh > .dockerignore 1`] = `
"*.less.json
*.log
*.tsbuildinfo
.vscode
coverage
dist
node_modules
"
`;
